### ISSUE TITLE
feature: option for validating only changed fields or a specified subset of fields 

### ISSUE THREAD

**Issue by captaincaius** [Automattic/mongoose#7421]
**Do you want to request a *feature* or report a *bug*?**

*feature*

**What is the current behavior?**

when using save() on a given object, validation is black-or-white - i.e. either all fields are validated (even unchanged ones) or none.

**If the current behavior is a bug, please provide the steps to reproduce.**

The use case:
Situations when you don't want to collect ALL field values for a document all in one place (e.g. multi-step forms, etc).  Especially in the case of onboarding a user, this poses a problem, because some of a document's other required fields may not even be set while working on that particular subset of fields, but you want to persist their progress to the database along the way.

**What is the expected behavior?**

The ability to tell save() to either:
- only validate updated fields
- only validate a specified subset of fields

**Workarounds and why they're not ideal**

1. bypass validation entirely: you WANT to use the power of the validation for the fields you are saving
2. use update instead of save: you lose hooks - plus, working at this level of abstraction is part of what makes mongoose awesome.
3. don't persist to the database along the way: users :-/ plus the extra work needed on the client-side.
4. split up the schema: IMO the data model shouldn't be driven by UX decisions as it makes changes a lot harder

**Please mention your node.js, mongoose and MongoDB version.**
node 10 / mongoose 5 / mongo 3.2

**Comment by vkarpov15** [comment]
Not a bad idea. We support passing a list of paths to validateSync: https://mongoosejs.com/docs/api.html#document_Document-validateSync . But if you just want to work around making certain fields required depending on the user's status, you can use conditional required:

```
new Schema({
  status: String,
  approvedAt: { type: Date, required: function() { return this.status === 'APPROVED'; } }
});
```

Is that the problem you have or is there something else?

**Comment by captaincaius** [comment]
Hmm I think I get what you mean - add a field for each subset of fields that submit together and make all their validations dependent on whether that field is set.

Thanks for the suggestion!  I'll try it.  It does couple the schema to UX a bit which I was hoping I could avoid, but it's the cleanest option I've got right now it seems.

I initially suggested this because I noticed in the code that something very similar is happening with isSelected(), but you pointed out validateSync and it seems its approach would be simpler (i.e. get all paths then filter).

Would this be a welcome PR for a new contributor?  Doesn't seem hard but I wouldn't want to add review burden without asking first ;).
